packages/ember-glimmer-template-compiler/lib/plugins/transform-old-class-binding-syntax.js

```diff
@@ -0,0 +1,111 @@
+export default function TransformOldClassBindingSyntax(options) {
+  this.syntax = null;
+  this.options = options;
+}
+
+TransformOldClassBindingSyntax.prototype.transform = function TransformOldClassBindingSyntax_transform(ast) {
+  let b = this.syntax.builders;
+  let walker = new this.syntax.Walker();
+
+  walker.visit(ast, node => {
+    if (!validate(node)) { return; }
+
+    let allOfTheMicrosyntaxes = [];
+    let allOfTheMicrosyntaxIndexes = [];
+    let classPair;
+
+    each(node.hash.pairs, (pair, index) => {
+      let { key } = pair;
+
+      if (key === 'classBinding' || key === 'classNameBindings') {
+        allOfTheMicrosyntaxIndexes.push(index);
+        allOfTheMicrosyntaxes.push(pair);
+      } else if (key === 'class') {
+        classPair = pair;
+      }
+    });
+
+    if (allOfTheMicrosyntaxes.length === 0) { return; }
+
+    let classValue = [];
+
+    if (classPair) {
+      classValue.push(classPair.value);
+      classValue.push(b.string(' '));
+    } else {
+      classPair = b.pair('class', null);
+      node.hash.pairs.push(classPair);
+    }
+
+    each(allOfTheMicrosyntaxIndexes, index => {
+      node.hash.pairs.splice(index, 1);
+    });
+
+    each(allOfTheMicrosyntaxes, ({ value, loc }) => {
+      let sexprs = [];
+      // TODO: add helpful deprecation when both `classNames` and `classNameBindings` can
+      // be removed.
+
+      if (value.type === 'StringLiteral') {
+        let microsyntax = parseMicrosyntax(value.original);
+
+        buildSexprs(microsyntax, sexprs, b);
+
+        classValue.push.apply(classValue, sexprs);
+      }
+    });
+
+    let hash = b.hash();
+    classPair.value = b.sexpr(b.path('concat'), classValue, hash);
+  });
+  return ast;
+};
+
+function buildSexprs(microsyntax, sexprs, b) {
+  for (let i = 0; i < microsyntax.length; i++) {
+    let [propName, activeClass, inactiveClass] = microsyntax[i];
+    let sexpr;
+
+    // :my-class-name microsyntax for static values
+    if (propName === '') {
+      sexpr = b.string(activeClass);
+    } else {
+      let params = [b.path(propName)];
+
+      if (activeClass || activeClass === '') {
+        params.push(b.string(activeClass));
+      } else {
+        params.push(b.string(propName));
```

Comment:
I think this needs to be split over `.`, and if you want to go this route, you would have to do the dasherize here, or otherwise distinguish that from the `foo:barBaz` case where `barBaz` should not be dasherized (I believe?)